### ISSUE TITLE
ZERO 3W build failure

### ISSUE THREAD

**Issue by elockman** [radxa/meta-rockchip#21]
It appears that the **ZERO 3W** should be supported, but there are some build errors that reveal lost upstream commits, and a device tree error.

I am building on **Ubuntu 22.04.5 LTS** with the following command:
`MACHINE=radxa-zero-3w bitbake core-image-minimal
`

First, the kernel tag is missing for "next-20240611" so I updated the SRCREV
```
- # this is tag 'next-20240611'
- SRCREV = "a957267fa7e9159d3d2ee1421359ebf228570c68"
+ # this is tag 'v6.10'
+ SRCREV = "0c3836482481200ead7b416ca80c68a29cfdaabd"
```
Second, the u-boot commit wasn't found, so I updated the SRCREV
```
- SRCREV:radxa-zero-3 = "8cdf606e616baa36751f3b4adcfaefc781126c8c"
- SRCREV:radxa-zero-3:rk-u-boot-env = "8cdf606e616baa36751f3b4adcfaefc781126c8c"
+ SRCREV:radxa-zero-3 = "b6c8629b75593907cea014a1f440287eeab32631"
+ SRCREV:radxa-zero-3:rk-u-boot-env = "b6c8629b75593907cea014a1f440287eeab32631"
```
Finally, I am getting the following error:
```
|   SYNC    include/config/auto.conf
|   GEN     Makefile
| make[3]: *** No rule to make target 'arch/arm64/boot/dts/rockchip/rk3566-radxa-zero-3w.dtb'.  Stop.
| make[2]: *** [/home/builder/projects/zero3w/yocto/build/tmp-glibc/work-shared/radxa-zero-3w/kernel-source/scripts/Makefile.build:485: arch/arm64/boot/dts/rockchip] Error 2
| make[1]: *** [/home/builder/projects/zero3w/yocto/build/tmp-glibc/work-shared/radxa-zero-3w/kernel-source/Makefile:1398: rockchip/rk3566-radxa-zero-3w.dtb] Error 2
| make: *** [/home/builder/projects/zero3w/yocto/build/tmp-glibc/work-shared/radxa-zero-3w/kernel-source/Makefile:240: __sub-make] Error 2
| ERROR: oe_runmake failed
| WARNING: exit code 1 from a shell command.
ERROR: Task (/home/builder/projects/zero3w/yocto/build/../meta-rockchip/recipes-kernel/linux/linux-torvalds-next_git.bb:do_compile) failed with exit code '1'
```

Here is the log file:
[log_do_compile.txt](https://github.com/user-attachments/files/17467051/log_do_compile.txt)

Any help is appreciated. If I can get this the build, I'll be getting some boards for development.

**Comment by elockman** [comment]
@RadxaStephen @RadxaYuntian 
The ZERO 3W project is broken. Can you help with getting this project building again? See above for details on the errors I've run into. Thanks.

**Comment by RadxaYuntian** [comment]
Which branch you are using?

**Comment by elockman** [comment]
I'm using scarthgap. Please point me to another branch if it is a better option.

**Comment by elockman** [comment]
Making these changes helped build the image.
Linux kernel version 6.10 does not contain the ZERO 3W dts file, so I moved up to v6.11.
Alternatively, you could patch the dts file into another kernel version, but this was easier.

**/meta-rockchip/recipes-kernel/linux/linux-torvalds-next_git.bb**
```
- LINUX_VERSION = "6.10-rc3"
+ LINUX_VERSION = "6.11"
```

```
- # this is tag 'next-20240611'
- SRCREV = "a957267fa7e9159d3d2ee1421359ebf228570c68"
+ # this is tag 'v6.11'
+ SRCREV = "98f7e32f20d28ec452afb208f9cffc08448a2652"
```

**Comment by elockman** [comment]
Issue resolved by moving to Linux version 6.11.
